src/pow.rs

```diff
@@ -270,3 +270,26 @@ fn count_leading_zeros(data: &[u8]) -> u8 {
 
     leading_zeros
 }
+
+pub fn calculate_difficulty(
```

Comment:
Can you please add a very detailed explanation on the idea behind the motivation and implementation details of the formula as a docstring.

This is awesome by the way.